File: backend/src/main/java/com/vollailelink/backend/dto/PriceHistoryDTO.java
package com.vollailelink.backend.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.OffsetDateTime;

/**
 * Historique d'une variation de prix (CDC section 2.5).
 *
 * `prixAchat` n'apparait que dans les DTO admin : le prix d'achat du
 * producteur ne doit jamais etre expose publiquement (CDC 2.1).
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class PriceHistoryDTO {
    private Long id;
    private Long productId;
    private String productNom;
    private BigDecimal price;
    private Long changedBy;
    private String changedByPhone;
    private OffsetDateTime changedAt;
    private String reason;
}

File: frontend/projects/admin/src/app/dto/price-history.dto.ts
/**
 * Variation de prix historisée.
 * `reason` porte le motif saisi par l'administrateur, obligatoire côté serveur
 * pour que la variation soit traçable (CDC §2.5).
 */
export interface PriceHistoryDTO {
  id: number;
  productId?: number;
  productNom?: string;
  price: number;
  changedBy?: number;
  changedByPhone?: string;
  changedAt?: string;
  reason?: string;
}
